internal/handlers/handler.go

```diff
@@ -153,77 +150,31 @@ func (h *Handler) handleOpMsg(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg
 
 	h.metrics.requests.WithLabelValues(wire.OP_MSG.String(), cmd).Inc()
 
-	switch cmd {
-	case "buildinfo":
-		return h.shared.MsgBuildInfo(ctx, msg)
-	case "collstats":
-		return h.shared.MsgCollStats(ctx, msg)
-	case "create":
-		return h.shared.MsgCreate(ctx, msg)
-	case "dbstats":
-		return h.shared.MsgDBStats(ctx, msg)
-	case "drop":
-		return h.shared.MsgDrop(ctx, msg)
-	case "dropdatabase":
-		return h.shared.MsgDropDatabase(ctx, msg)
-	case "getcmdlineopts":
-		return h.shared.MsgGetCmdLineOpts(ctx, msg)
-	case "getlog":
-		return h.shared.MsgGetLog(ctx, msg)
-	case "getparameter":
-		return h.shared.MsgGetParameter(ctx, msg)
-	case "hostinfo":
-		return h.shared.MsgHostInfo(ctx, msg)
-	case "ismaster", "hello":
-		return h.shared.MsgHello(ctx, msg)
-	case "listcollections":
-		return h.shared.MsgListCollections(ctx, msg)
-	case "listdatabases":
-		return h.shared.MsgListDatabases(ctx, msg)
-	case "ping":
-		return h.shared.MsgPing(ctx, msg)
-	case "whatsmyuri":
-		return h.shared.MsgWhatsMyURI(ctx, msg)
-	case "serverstatus":
-		return h.shared.MsgServerStatus(ctx, msg)
-
-	case "createindexes", "delete", "find", "insert", "update", "count":
-		storage, err := h.msgStorage(ctx, msg)
-		if err != nil {
-			return nil, lazyerrors.Error(err)
-		}
+	if cmd == "listcommands" {
+		return SupportedCommands(ctx, msg)
+	}
 
-		switch cmd {
-		case "createindexes":
-			return storage.MsgCreateIndexes(ctx, msg)
-		case "delete":
-			return storage.MsgDelete(ctx, msg)
-		case "find", "count":
-			return storage.MsgFindOrCount(ctx, msg)
-		case "insert":
-			return storage.MsgInsert(ctx, msg)
-		case "update":
-			return storage.MsgUpdate(ctx, msg)
-		default:
-			panic("not reached")
+	if cmd, ok := commands[cmd]; ok {
+		if cmd.handler != nil {
+			return cmd.handler(h, ctx, msg)
+		} else {
+			storage, err := h.msgStorage(ctx, msg)
```

Comment:
```suggestion
		storage, err := h.msgStorage(ctx, msg)
```

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow